numpy/lib/npyio.py

```diff
@@ -1001,7 +1210,10 @@ def flatten_dtype_internal(self, dt):
     def pack_items(self, items, packing):
         """Pack items into nested lists based on re-packing info."""
         if packing is None:
-            return items[0]
+            if len(items) == 0:
+                return items
+            else:
+                return items[0]
```

Comment:
This doesn't look right to me - I think if `len(items) == 0`, the caller should have somehow ensured that `packing == list`.

Reply:
Ah, right. This is related to my second API question, about how `usecols=[]` should be handled.